### PR TITLE
Send data to CODAP in smaller chunks to avoid performance problems

### PR DESCRIPTION
**Pull request by pjanik**
[#157605204]

### CODE REVIEW

File: src/lab/import-export/codap-interface.js [hunk 1]
```diff
@@ -172,25 +184,33 @@ define(function(require) {
...
-          // Open case table after successful export.
-          this.openTable();
+        // If time series attributes are not defined, we will insert just a single item with per-run attributes only.
+        var values = (shouldExportTimeSeries ? timeSeriesData : [ null ]).map(function (data) {
+          var item = {};
+          perRunAttrs.forEach(function (attr, idx) {
+            item[attr.name] = perRunData[idx];
+          });
+          timeSeriesAttrs.forEach(function (attr, idx) {
+            item[attr.name] = data[idx];
```
Comment (kswenson) [inline src/lab/import-export/codap-interface.js]: It took me a minute to convince myself that the explicit null passed to `map()` above would never be dereferenced here. A comment to that effect would be helpful.

**Comment by kswenson** [comment]
@jsandoe You should take a look at this PR. Perhaps a suggestion in the plugin API docs about breaking up the data into chunks would make sense.

File: src/lab/import-export/codap-interface.js (hunk 1 above)
Comment (pjanik) [reply src/lab/import-export/codap-interface.js]: You're right, this is not very clear. I've updated comments above and repalced null with some bogus string to make it clearer that it's gonna be ignored. Thanks!

**Comment by pjanik** [comment]
Thanks @kswenson. @jsandoe, yes, note in the API docs would be helpful. Actually, I think that it would be great if the handling of large arrays was implented on the CODAP side. It's not a big deal for a client to send thousands of points using postMessage, so batching here seems a bit strange. It's CODAP having some troubles with that. Also, we could avoid reimplementing similar workarounds in various plugins.

**Pull request merged**